### PR TITLE
Lint Kotlin code in buildSrc

### PR DESCRIPTION
**Pull request by eskatos**
Using the experimental [org.gradle.kotlin.ktlint-convention](https://plugins.gradle.org/plugin/org.gradle.kotlin.ktlint-convention) convention plugin. Its source can be found in the [`:plugins-experiments`](https://github.com/gradle/kotlin-dsl/tree/develop/plugins-experiments) project of the `gradle/kotlin-dsl` repository. Linting tasks are cacheable, see [here](https://github.com/gradle/kotlin-dsl/blob/5f1740de6e3129f1b37120243d9afe682405fe77/plugins-experiments/src/test/kotlin/org/gradle/kotlin/dsl/experiments/plugins/GradleKotlinDslKtlintConventionPluginTest.kt#L31) and [here](https://scans.gradle.com/s/hgowxnomgosky/timeline?cacheableFilter=CACHEABLE&typeFilter=org.gradle.api.tasks.JavaExec).

This pull-request applies the plugin to `buildSrc` sub projects that include Kotlin sources and then fix all linting errors.

### CODE REVIEW

File: buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt [hunk 1]
```diff
@@ -27,18 +28,26 @@ class DefaultJavaInstallation(val current: Boolean, private val javaHome: File)
...
 
+
 private
 const val java7HomePropertyName = "java7Home"
+
+
 private
 const val testJavaHomePropertyName = "testJavaHome"
+
+
```
Comment (wolfs) [inline buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt]: Why two newlines between all those declarations? Is this somehow Kotlin std?

Comment (eskatos) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt]: It is the `gradle/kotlin-dsl` convention, see https://github.com/gradle/kotlin-dsl/blob/5f1740de6e3129f1b37120243d9afe682405fe77/plugins-experiments/src/test/kotlin/org/gradle/kotlin/dsl/experiments/plugins/GradleKotlinDslKtlintConventionPluginTest.kt#L115-L179

This, like other rules enforced by this convention, is what we've been doing in `gradle/kotlin-dsl`.

Comment (eskatos) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt]: Also see https://github.com/gradle/kotlin-dsl/pull/750 for more info

Comment (wolfs) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt]: So top level declarations should be separated by two lines.
I still don't understand why we are doing this...
Probably mostly for not having anything similar in Java/Groovy/Checkstyle

Comment (bamboo) [reply buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt]: @wolfs it's our experience with the `kotlin-dsl` codebase that this makes for more readable sources as Kotlin files tend to have many top-level declarations.

**Pull request merged**